core-model/src/main/java/com/sequenceiq/cloudbreak/domain/stack/loadbalancer/aws/AwsTargetGroupArns.java

```diff
@@ -0,0 +1,27 @@
+package com.sequenceiq.cloudbreak.domain.stack.loadbalancer.aws;
+
+/**
+ * Contains the ARNs for both a listener and the target group that listener forwards traffic to.
+ */
+public class AwsTargetGroupArns {
```

Comment:
I don't disagree that there's a lot of hoops to jump though. In fact, there should be exactly the three you mentioned, so I'm not sure what the fourth hoop is. If you can point it out specifically I can try to remove it.

We're mostly running into dependency problems when it comes to object optimization. We need to convert AWS metadata -> database object -> response object. So we do `AwsLoadBalancerMetadata` (metadata object) -> `AwsLoadBalancerConfig` (database object) -> `AwsLoadBalancerSettings` (response object). One possible optimization might be to just use the DB classes directly instead of the `CloudLoadBalancerMetadata` object. But the problem is, the metadata collector modules don't have a dependency on the database entity module, so I can't create objects of that type there. Similarly, the database entity modeul doesn't have a dependency on the metadata collector modules, so I can't create an object in those modules and use it directly as part of the database entries. In these various objects seem to meet in the `core` module, which does have the appropriate dependencies to use all the objects. But we need a way to get the information to `core`, which is where the different types of objects come from. They're originated in parts of the code that can't share information, so we need a module specific way to propagate that information back to the `core` module that can put it all together.

Man, the names of these things are still terrible. I'm going to try again.

Reply:
Okay:
* Metadata object: `LoadBalancerMetadataBase` for base object, `AwsLoadBalancerMetadata` for specific implementation (unchanged)
* Database object: `LoadBalancerConfigDbWrapper` for base object, `AwsLoadBalancerConfigDb` for specific object. All objects the ultimate end up in the database now have `Db` as part of their name.
* Response object: `AwsLoadBalancerResponse`
Now hopefully it's actually clear what part each objects belongs to from the name.

Reply:
I understand now. makes sense and the direction dependencies flow in is probably the only way to do it.